### PR TITLE
ci: Change versioning pipeline

### PR DESCRIPTION
**Pull request by grianbrcom**
GH action for changelog draft generation was added.
GH action for
releasing was added.
Jenkinsfile changed to use branch and build number
in package version.

Signed-off-by: Anton Grigorev <anton.grigorev@broadcom.com>

### CODE REVIEW

**Review by temanbrcom — APPROVED** [review]
🚘

File: release/version-extract.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const fs = require('fs');
```
Comment (zacanbrcom) [inline release/version-extract.js:1]: do we need license header here?

Comment (grianbrcom) [reply release/version-extract.js:1]: Yes, you are right.

File: server/pom.xml [hunk 2]
```diff
@@ -47,12 +47,11 @@
         <maven.coveralls.plugin.version>4.3.0</maven.coveralls.plugin.version>
         <maven.jacoco.plugin.version>0.8.5</maven.jacoco.plugin.version>
         <maven.resources.plugin.version>3.0.2</maven.resources.plugin.version>
-        <maven.source.plugin.version>3.0.1</maven.source.plugin.version>
         <maven.surefire.plugin.version>3.0.0-M3</maven.surefire.plugin.version>
         <mockito.core.version>2.23.0</mockito.core.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-        <revision>0.13.0+NEXT</revision>
+        <revision>1.0.0-SNAPSHOT</revision>
```
Comment (filipkroupa) [inline server/pom.xml:54]: Does this mean the server code won't be versioned in the repository anymore? Version would be only put there at build time?

Comment (grianbrcom) [reply server/pom.xml:54]: We decided not to version our java part. We don't expose our jar in public and use it only internally. So the end-user in not care what exact java part version he have. And also no one has a way to download some particular java part version separately.
That's why we decided to shrink our effort to keep all versions updated.

Comment (filipkroupa) [reply server/pom.xml:54]: 👍 🆗

**Review by zacanbrcom — APPROVED** [review]
let's ship it! 🚢

**Review by asatklichov — APPROVED** [review]
💯

**Pull request merged**